projects/extension/src/containers/Popup.tsx

```diff
@@ -1,15 +1,72 @@
-import React, { FunctionComponent, useEffect, useRef, useState } from "react"
+import React, {
+  FunctionComponent,
+  ReactNode,
+  useEffect,
+  useRef,
+  useState,
+} from "react"
 
 import { MdOutlineSettings, MdCallMade } from "react-icons/md"
+import { BiDotsHorizontalRounded } from "react-icons/bi"
 
-import { Logo, ConnectedTab } from "../components"
+import { Accordion, ConnectedTab, Logo } from "../components"
 import { Background } from "../background"
-import { TabInterface } from "../types"
+
+const knownChains = ["polkadot", "kusama", "westend", "rococo"]
+
+interface PopupChains {
+  chainName: string
+  details: ChainDetails[]
+}
+
+interface ChainDetails {
+  tabId?: number
+  url?: string
+  peers: number
+  isSyncing: boolean
+  chainId: string
+}
 
 const Popup: FunctionComponent = () => {
-  const disconnectTabRef = useRef<(tapId: number) => void>((_: number) => {})
-  const [activeTab, setActiveTab] = useState<TabInterface | undefined>()
-  const [otherTabs, setOtherTabs] = useState<TabInterface[]>([])
+  const disconnectTab = useRef<(tabId: number) => void>((_: number) => {})
+  const [connChains, setConnChains] = useState<PopupChains[] | undefined>()
+
+  const refresh = () => {
+    chrome.runtime.getBackgroundPage((backgroundPage) => {
+      const bg = backgroundPage as Background
+      const allChains: PopupChains[] = []
+
+      bg.uiInterface.chains.forEach((c) => {
+        const i = allChains.findIndex((i) => i.chainName === c.chainName)
+        if (i === -1) {
+          allChains.push({
+            chainName: c.chainName,
+            details: [
+              {
+                tabId: c.tab?.id,
+                url: c.tab?.url,
+                peers: c.peers,
+                isSyncing: c.isSyncing,
+                chainId: c.chainId,
+              },
+            ],
+          })
+        } else {
+          const details = allChains[i]?.details
+          if (!details?.map((d) => d.tabId).includes(c.tab?.id)) {
+            details?.push({
+              tabId: c.tab?.id,
+              url: c.tab?.url,
+              peers: c.peers,
+              isSyncing: c.isSyncing,
+              chainId: c.chainId,
+            })
+          }
```

Comment:
You know that `i` is a valid index, so you don't need to handle this weird "null" situation
```suggestion
          const details = allChains[i]!.details
          if (!details.map((d) => d.tabId).includes(c.tab?.id)) {
            details.push({
              tabId: c.tab?.id,
              url: c.tab?.url,
              peers: c.peers,
              isSyncing: c.isSyncing,
              chainId: c.chainId,
            })
          }
```